05-Introducing-Rigid-Body-Molecular-Dynamics/03-Thermodynamic-Quantities-of-Rigid-Bodies.ipynb

```diff
@@ -0,0 +1,354 @@
+    "\n",
+    "m = 2\n",
+    "N_particles = 2 * m**3\n",
+    "spacing = 10.\n",
+    "K = math.ceil(N_particles**(1 / 3))\n",
+    "L = K * spacing\n",
+    "x = numpy.linspace(-L / 2, L / 2, K, endpoint=False)\n",
+    "position = list(itertools.product(x, repeat=3))\n",
+    "position = numpy.array(position)\n",
+    "position[:,0] = position[:,0]\n",
+    "snapshot = gsd.hoomd.Snapshot()\n",
+    "snapshot.particles.N = N_particles\n",
+    "snapshot.particles.position = position[0:N_particles,:]\n",
+    "snapshot.particles.orientation = [(1,0,0,0)] * N_particles\n",
+    "snapshot.particles.typeid = [0] * N_particles\n",
+    "snapshot.configuration.box = [L, L, L, 0, 0, 0]\n",
+    "snapshot.particles.mass = [Mass_Rigid_body] * N_particles\n",
+    "snapshot.particles.types = ['rigid_body','constituent_sphere']\n",
+    "snapshot.particles.moment_inertia = [(Inertia_Tensor[0,0],Inertia_Tensor[1,1],Inertia_Tensor[2,2])] * N_particles\n",
+    "snapshot.particles.angmom = [(1,1,1,1)] * N_particles\n",
+    "with gsd.hoomd.open(name='lattice_rigid.gsd', mode='wb') as f:\n",
+    "    f.append(snapshot)\n",
+    "\n",
+    "temperature = 1.\n",
+    "\n",
+    "cpu = hoomd.device.CPU()\n",
+    "sim = hoomd.Simulation(device=cpu, seed=1)\n",
+    "sim.create_state_from_gsd(filename='lattice_rigid.gsd')\n",
+    "\n",
+    "epsilon = 1\n",
+    "sigma = 2/2**(1/6)\n",
+    "integrator = hoomd.md.Integrator(dt=0.0001,integrate_rotational_dof=True)\n",
+    "cell = hoomd.md.nlist.Cell(buffer=0,exclusions = ['body'])\n",
+    "lj = hoomd.md.pair.LJ(nlist=cell)\n",
+    "lj.params[('rigid_body', 'rigid_body')] = dict(epsilon=0, sigma=sigma)\n",
+    "lj.params[('rigid_body', 'constituent_sphere')] = dict(epsilon=0, sigma=sigma)\n",
+    "lj.params[('constituent_sphere', 'constituent_sphere')] = dict(epsilon=epsilon, sigma=sigma)\n",
+    "lj.r_cut[('constituent_sphere', 'constituent_sphere')] = 4 * sigma \n",
+    "lj.r_cut[('rigid_body', 'rigid_body')] = 0.\n",
+    "lj.r_cut[('rigid_body', 'constituent_sphere')] = 0.\n",
+    "integrator.forces.append(lj)\n",
+    "\n",
+    "rigid_centers_and_free_filter = hoomd.filter.Rigid((\"center\", \"free\"))\n",
+    "nvt = hoomd.md.methods.NVT(kT = temperature, tau = 1., filter = rigid_centers_and_free_filter)\n",
+    "integrator.methods.append(nvt)\n",
+    "\n",
+    "rigid = constrain.Rigid()\n",
+    "constituent_types = ['constituent_sphere','constituent_sphere']\n",
+    "rigid.body['rigid_body'] = {\n",
+    "    \"constituent_types\": constituent_types,\n",
+    "    \"positions\":local_position,\n",
+    "    \"orientations\":[(1,0,0,0)] * len(constituent_types),\n",
+    "    \"charges\":[0.] * len(constituent_types),\n",
+    "    \"diameters\":[2.] * len(constituent_types)}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "fc46a9ba-34ee-4750-9a5c-be492178c155",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "rigid.create_bodies(sim.state)\n",
```

Comment:
This is the first place you show this function, and you don't even really describe it.